sdk/avalon_sdk/fabric/fabric_wrapper.py

```diff
@@ -0,0 +1,168 @@
+
+
+class FabricWrapper():
+    """
+    Fabric wrapper class to interact with fabric blockchain
+    It provides wrapper functions to invoke and query chain code.
+    """
+
+    def __init__(self, config):
+        """
+        Constructor to initialize wrapper with required parameter
+        Params:
+            config is dict containing parameters for fabric.
+            These parameters are read from toml file.
+        """
+        if self.__validate(config):
+            # Read network file path from fabric toml file.
+            with open(self.__network_conf_file, 'r') as profile:
+                self.__network_config = json.load(profile)
+            self.__channel_name = config["fabric"]["channel_name"]
+            self.__orgname = base.get_net_info(self.__network_config,
+                                               'client', 'organization')
+            logging.info("Org name choose: {}".format(self.__orgname))
+            self.__peername = random.choice(base.get_net_info(
+                self.__network_config, 'organizations', self.__orgname,
+                'peers'))
+            # Get a txn committer
+            self.__txn_committer = tx_committer.TxCommitter(
+                self.__network_conf_file,
+                self.__channel_name, self.__orgname,
+                self.__peername, 'Admin')
+            with open(self.__tcf_home + '/sdk/avalon_sdk/fabric/' +
+                      'methods.json', 'r') as all_methods:
+                self.__valid_calls = json.load(all_methods)
+        else:
+            raise Exception("Invalid configuration parameter")
+
+    def __validate(self, config):
+        """
+        validates parameter from config parameters for existence.
+        Returns false if validation fails and true if it success
+        """
+        if config["fabric"]["fabric_network_file"] is None:
+            logging.error("Fabric network configuration is not set!!")
+            return False
+        self.__tcf_home = environ.get("TCF_HOME", "../../../")
+        self.__network_conf_file = self.__tcf_home + "/" + \
+            config["fabric"]["fabric_network_file"]
+        if not exists(self.__network_conf_file):
+            raise FileNotFoundError(
+                "File not found at path: {0}".format(
+                    realpath(__network_conf_file)))
+        if config["fabric"]["channel_name"] is None:
+            logging.error("Channel name is not specified")
+            return False
+        return True
+
+    def invoke_chaincode(self, chaincode_name, method_name, params):
+        """
+        This is wrapper method to invoke the chain code.
+        Params:
+            chaincode_name is name of the chain code
+            method_name is chain code method name
+            params is list of arguments to method
+        Returns:
+            If the call to chain code query then it
+            returns the payload of chain code response
+            on success or None on error.
```

Comment:
This function is too complex. Could be split into two - query chain code and invoke chain code which results in state change. May be in separate PR